README.md

```diff
@@ -246,6 +246,25 @@ This will poll in the defined interval and only resolve with a TokenSet once one
 will handle the defined `authorization_pending` and `slow_down` "soft" errors and continue polling
 but upon any other error it will reject. With tokenSet received you can throw away the handle.
 
+### Client Credentials Grant Flow
+
+Exchanging the client_id and client_secret of your application for an access token, to authenticate
+at a third-party api. Not on behalf of a user. Suitable for Machine-to-Machine authentication.
+
+**See the [documentation](./docs/README.md#clientgrantbody-extras) for full API details.**
+
+```js
+const client = new googleIssuer.Client({
+  client_id: 'zELcpfANLqY7Oqas',
+  client_secret: 'TQV5U29k1gHibH5bx1layBo0OSAvAbRT3UYW3EWrSYBB5swxjVfWUa1BS8lqzxG/0v9wruMcrGadany3',
+});
+
+const tokenSet = await client.grant({
+  audience: 'insert-your-audience',
```

Comment:
```suggestion
  resource: 'urn:example:third-party-api',
```

Reply:
Looking into the auth0 web ui, the thing that jumps into view multiple times is the audience, and the quickstart shows it with an audience.

I tried to find something about this resource string, but couldn't. I also tried to look into the code, but the type of the grant function is just

```ts
export interface GrantBody {
  grant_type: string;

  [key: string]: unknown;
}
```

And resource is never mentioned in the docs of node-openid-client.

So I have no idea what the resource string is, and I definitely would want to see where to put the audience in the example.

Reply:
auth0 mentions an urn in the context of SAML: https://auth0.com/blog/url-uri-urn-differences/

the wikipedia entry says SAML is important for SSO: https://en.wikipedia.org/wiki/Security_Assertion_Markup_Language

Is this really relevant for the client credentials flow?

Reply:
https://datatracker.ietf.org/doc/html/rfc8707

Reply:
@sezanzeb This repo's history will show you that it's aim is to support the OIDC Conforming behaviour as RP. 

That means couple of things:

- SHOULD promote conforming use cases, like suggesting RFC 8707 way of asking for a resource when those OPs implement RFC 8707.
- Should NOT promote any edge cases for different OP implementations around. 
Auth0 is OIDC provider vendor, but it's not conforming to 8707 spec as described. 
For example, I may not be using Auth0 as my OP, I may or may not be using `grant` function, and I don't want to be mislead into having to think about audience parameter at all in those cases.
So that's a rationale why `audience` should not land in docs in this library. More like **in your server implementation repo is where the place for those docs is.**
- Should NOT prevent you from passing extra parameters when those are required. It's also stated in types with this type descriptor: `[key: string]: unknown;`, that's why typescript is not complaining when you add audience. Also, you can look [at source code ](https://github.com/panva/node-openid-client/blob/main/lib/client.js#L1341) and see that all arguments you define are in fact being passed to provider.

Reply:
> Should NOT promote any edge cases for different OP implementations around.

This is for giving hints for less knowledgeable people like me. I'm not a specialist for oauth, and my intention for using a library is, that it will help me set things up correctly without having to have a deeper understanding of the thing at hand.

Actually, I don't remember where I even got the idea that I can insert audience there. It might have been that I just confused `AuthorizationParameters` with `DeviceAuthorizationParameters` when looking into the code, and therefore by pure luck correctly inserted `audience` into the `grant` function...

I made another commit, this time removing audience from README.md, and saying that audience is non-standard in docs/README.md